roles/validate/files/rules/required_rules/201_global_spanning_tree.py

```diff
@@ -0,0 +1,35 @@
+class Rule:
+    id = "201"
+    description = "Verify a spanning tree protocol mutually exclusive parameters."
+    severity = "HIGH"
+
+    @classmethod
+    def match(cls, inventory):
+        results = []
+        if inventory.get("vxlan", None):
+            if inventory["vxlan"].get("global", None):
+                if inventory["vxlan"].get("global", None).get("spanning_tree", None):
+                    root_bridge_protocol = inventory["vxlan"]["global"]["spanning_tree"].get("root_bridge_protocol", None)
+                    vlan_range = inventory["vxlan"]["global"]["spanning_tree"].get("vlan_range", None)
+                    mst_instance_range = inventory["vxlan"]["global"]["spanning_tree"].get("mst_instance_range", None)
+
+                    if vlan_range and mst_instance_range:
+                        results.append(
+                            "vxlan.global.spanning_tree.vlan_range and vxlan.global.spanning_tree.mst_instance_range "
+                            "both cannot be configured at the same time. Please choose one depending on the "
+                            "vxlan.global.spanning_tree.root_bridge_protocol selected."
+                        )
+
+                    if root_bridge_protocol == "rpvst+" and not vlan_range:
+                        results.append(
+                            "vxlan.global.spanning_tree.vlan_range must be set when the "
+                            "spanning tree vxlan.global.spanning_tree.root_bridge_protocol is set to rpvst+."
+                        )
+
+                    if root_bridge_protocol == "mst" and not mst_instance_range:
+                        results.append(
```

Comment:
Seems like this message should be similar to the message above on line `25,26`.

`vxlan.global.spanning_tree.mst_instance_range must be set when the spanning tree vxlan.global.spanning_tree.root_bridge_protocol is set to mst`

Reply:
I removed the latter two rule checks as they were unnecessary since we will pull from defaults if the user doesn't specify anything.